lib/note-list/no-notes.tsx

```diff
@@ -1,43 +1,62 @@
 import React from 'react';
 import { useDispatch, useSelector } from 'react-redux';
 
+import NotesIcon from '../icons/notes';
+import TagIcon from '../icons/tag';
+import TrashIcon from '../icons/trash';
+
 import actions from '../state/actions';
 
 import * as S from '../state';
 
 const NoNotes = () => {
   const hasLoaded = useSelector((state: S.State) => state.ui.hasLoadedNotes);
   const searchQuery = useSelector((state: S.State) => state.ui.searchQuery);
+  const showTrash = useSelector((state: S.State) => state.ui.showTrash);
+  const openedTag = useSelector((state: S.State) => state.ui.openedTag);
   const dispatch = useDispatch();
 
-  const getMessage = () => {
-    return hasLoaded
-      ? searchQuery.length
-        ? 'No Results'
-        : 'No Notes'
-      : 'Loading Notes';
-  };
-
-  const getButton = () => {
-    return searchQuery.length ? (
+  const getSearchButton = () => {
+    return (
       <button
         onClick={() =>
           dispatch(actions.ui.createNote({ content: searchQuery }))
```

Comment:
🤨 nothing like deep coupling - how'd that get in there?

Reply:
Yeah. I'm not sure if my approach to this refactoring of the empty note list views is the best way to go. Any thoughts would be much appreciated.

Reply:
best policy is to not change code styling in an unrelated PR so I wouldn't recommend doing anything differently. that being said I occasionally point out things in code review that are unrelated but which stand out to me. this is one of those because I saw `dispatch` inside a UI component, which is something that sets off all the alarms in my head